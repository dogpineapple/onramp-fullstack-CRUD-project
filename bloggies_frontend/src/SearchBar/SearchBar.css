.SearchBar {
  width: 20rem;
}