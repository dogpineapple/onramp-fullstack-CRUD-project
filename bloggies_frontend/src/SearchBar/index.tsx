import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import React, { useState } from "react";
import { Button,Form } from "react-bootstrap";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import { getSearchResultsFromAPI } from "../redux/actionCreators";
import "./SearchBar.css";
import { useHistory } from "react-router";
import { useDispatch } from "react-redux";

/**
 * `SearchBar` renders a form for entering a search term
 * to search for posts and users matching the term.
 * Dispatches an action for `getSearchResultsFromAPI` upon submission. 
 */
function SearchBar() {
  const [ formData, setFormData ] = useState({ searchTerm: "" });
  const history = useHistory();
  const dispatch = useDispatch();

  const handleSubmit = (evt: React.FormEvent<HTMLFormElement>) => {
    evt.preventDefault();
    dispatch(getSearchResultsFromAPI(formData.searchTerm));
    history.push(`/search?term=${formData.searchTerm}`);
  }

  const handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = evt.target;
    setFormData(currData => ({ ...currData, [name]: value }));
  }

  return (
    <div className="SearchBar">
      <Form className="d-flex" onSubmit={handleSubmit}>
          <Form.Control name="searchTerm" value={formData.searchTerm} placeholder="Search" onChange={handleChange} required/>
          <Button type="submit"><span><FontAwesomeIcon icon={faSearch}></FontAwesomeIcon></span></Button>
      </Form>
    </div>
  );
};

export default SearchBar;