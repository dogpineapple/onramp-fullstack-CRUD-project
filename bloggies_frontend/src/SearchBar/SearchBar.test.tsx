import React, { Component } from 'react';
import Enzyme, { mount, ReactWrapper, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import SearchBar from './index';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MOCK_STORE } from '../jest.mock';
Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

describe('SearchBar', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(shallow<Component>(
      <Provider store={mockStore(MOCK_STORE)}>
        <SearchBar />
      </Provider>
    ).get(0));
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});