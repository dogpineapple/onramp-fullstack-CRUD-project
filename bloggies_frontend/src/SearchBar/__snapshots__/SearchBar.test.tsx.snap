// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match snapshot 1`] = `
<SearchBar>
  <div
    className="SearchBar"
  >
    <Form
      className="d-flex"
      inline={false}
      onSubmit={[Function]}
    >
      <form
        className="d-flex"
        onSubmit={[Function]}
      >
        <FormControl
          name="searchTerm"
          onChange={[Function]}
          placeholder="Search"
          required={true}
          value=""
        >
          <input
            className="form-control"
            name="searchTerm"
            onChange={[Function]}
            placeholder="Search"
            required={true}
            value=""
          />
        </FormControl>
        <Button
          active={false}
          disabled={false}
          type="submit"
          variant="primary"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit"
          >
            <span>
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-search fa-w-16 "
                  data-icon="search"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
            </span>
          </button>
        </Button>
      </form>
    </Form>
  </div>
</SearchBar>
`;
